utils/s3/client.cc

```diff
@@ -157,6 +157,7 @@ shared_ptr<client> client::make(std::string endpoint, endpoint_config_ptr cfg, s
 }
 
 future<> client::update_credentials_and_rearm() {
+    _creds_provider_chain->invalidate_credentials();
```

Comment:
Does it mean that until the below `get_aws_credentials()` resolves (i.e. -- until server gives us new creds) the whole client lives without creds and waits for the server to give them back?